### PR TITLE
Setup axios for overall app.

### PR DESCRIPTION
**Pull request by su-sh**
#3 to be merged before this branch.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -52,13 +56,150 @@ a {
...
+.story-item {
+  width: calc(100% - 15px);
+}
+.pagination {
+<<<<<<< HEAD
+  padding: 0 5px 10px 5px;
+=======
+  padding: 10px 5px 10px 5px;
+>>>>>>> a6c56a01e972ff51fd627316b8dee02a1c856f1a
+  position: relative;
```
Comment (CuriousSugam) [inline src/App.css:198]: merge conflict not resolved!

File: src/api/api.js [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
  *
- *
+ * @returns {object}
  * @param {*} storiesType
  */
-export const getStoriesIndex = storiesType => {
-  return axios.get(`https://hacker-news.firebaseio.com/v0/${storiesType}.json`);
+export const getStoriesIndexArray = storiesType => {
+  return axios
+    .get(`${BASE_URL}${storiesType}.json`)
```
Comment (CuriousSugam) [inline src/api/api.js]: I would prefer the constant to be this way.
```
const STORY_TYPE_URL = `${BASE_URL}/${storiesType}.json`);

return axios
    .get(STORY_TYPE_URL)
   .then(...)
```

File: src/api/api.js [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+};
+
+/**
+ *
+ * @returns {promise}
+ * @param {string} id
+ */
+export const getItem = id => {
+  return axios
+    .get(`${BASE_URL}item/${id}.json`)
```
Comment (CuriousSugam) [inline src/api/api.js]: here too

File: src/component/Item.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    super();
+    this.state = {
+      type: undefined,
+      data: undefined
+    };
+  }
+
+  componentDidMount = async () => {
+    const itemId = this.props.match.params.id;
+
```
Comment (CuriousSugam) [inline src/component/Item.js]: remove this emptyline

File: src/component/Item.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      </div>
+    );
+  }
+
+}
+
+Item.propTypes = {
+  match: {
+    params: {
+      id: PropTypes.number
```
Comment (CuriousSugam) [inline src/component/Item.js]: Use PropTypes.shape

File: src/component/Loading.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import '.././loading.css';
```
Comment (CuriousSugam) [inline src/component/Loading.js]: `.././loading.css`? why do we need the single dot? Also separate imports with a line gap

File: src/component/Pagination.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import '../App.css';
+import PropTypes from 'prop-types';
```
Comment (CuriousSugam) [inline src/component/Pagination.js:2]:
```
import React from 'react';
import PropTypes from 'prop-types';

import '../App.css';
```

File: src/component/jobs/JobListItem.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  /**
+   *
+   *
+   * @returns {object}
+   * @memberof JobListItem
+   */
+  render() {
+    return !this.state.idLoaded ? (
+      <Loading />
+    ) : this.state.url ? (
```
Comment (CuriousSugam) [inline src/component/jobs/JobListItem.js]: there are two layers of nesting in the tertiary operator. try to simplify it. It is better to take the logic off the view.

**Review by CuriousSugam — CHANGES_REQUESTED** [review]

File: src/api/api.js [hunk 9]
```diff
@@ -1,10 +1,43 @@
...
+      return err;
+    });
+};
+
+/**
+ *
+ * @returns {promise}
+ * @param {string} id
+ */
+export const getItem = id => {
```
Comment (CuriousSugam) [inline src/api/api.js]: we use the term fetch when we make an api request to get the item. So, the function name `fetchItem` is preferrable.

File: src/component/jobs/JobsListWrapper.js [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+    };
+  }
+
+  /**
+   *
+   *
+   * @memberof Newest
+   */
+  componentDidMount = async () => {
+    const newArray = await getStoriesIndexArray(STORY_TYPE.JOB_STORIES);
```
Comment (CuriousSugam) [inline src/component/jobs/JobsListWrapper.js]: `fetch...`

File: src/component/jobs/JobsListWrapper.js [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+  };
+
+  /**
+   * This function checks if next button on pagination component is disabled.
+   *
+   * @param {number} currentPageNumber
+   * @returns {boolean}
+   * */
+  isDisabledRight = currentPageNumber => {
+    if ((currentPageNumber + 1) * 30 >= this.state.allStoriesIdList.length) {
```
Comment (CuriousSugam) [inline src/component/jobs/JobsListWrapper.js]: remove the calculation part from here. Make the condition of the if() statement more readable

File: src/component/jobs/JobsListWrapper.js [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+
+  /**
+   * This function checks if previous button on pagination component is disabled.
+   *
+   * @param {number} currentPageNumber
+   * @returns {boolean}
+   * */
+  isDisabledLeft = currentPageNumber => {
+    if (currentPageNumber === 0) {
+      return true;
```
Comment (CuriousSugam) [inline src/component/jobs/JobsListWrapper.js]: We can just return `currentPageNumber === 0`.

File: src/component/storyComment/Comment.js [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+            <img alt="up-img" src={upImg} />
+          </div>
+          <div className="left">{this.state.by}</div>
+          <div className="left">{getTimeDifference(this.state.time)}</div>
+        </div>
+
+        <div className="comment-bottom clearfix">
+          <div
+            className="comment-text"
+            dangerouslySetInnerHTML={{ __html: this.state.text }}
```
Comment (CuriousSugam) [inline src/component/storyComment/Comment.js:79]: Why do we need to use `dangerouslySetInnerHTML` here?

File: src/utils/utils.js [hunk 14]
```diff
@@ -0,0 +1,79 @@
+/**
+ *
+ * This function return sliced array of id's according to current page number.
+ *
+ * @param {array} array
+ * @param {*} pageSize
+ * @param {*} pageNumber
```
Comment (CuriousSugam) [inline src/utils/utils.js]: can pageNumber be a string?

**Review by CuriousSugam — CHANGES_REQUESTED** [review]

File: src/component/storyComment/Comment.js (hunk 13 above)
Comment (su-sh) [reply src/component/storyComment/Comment.js:79]: As comment text ie.[this.state.text] may contain multiple HTML tags within it, dangerouslySetInnerHTML was used.

File: src/component/jobs/JobsListWrapper.js [hunk 15]
```diff
@@ -132,35 +131,32 @@ class JobsListWrapper extends Component {
...
+          <Loading />
+        ) : (
+          this.state.showStoryIdList.map((storyId, index) => {
+            return (
+              <JobListItem
+                position={getPosition(index, this.state.currentPageNumber)}
+                key={storyId}
+                id={storyId}
+              />
+              // <div>asdf</div>
```
Comment (CuriousSugam) [inline src/component/jobs/JobsListWrapper.js:144]: remove this comment

**Review by CuriousSugam — APPROVED** [review]

**Pull request merged**